hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/AbstractContainerReportHandler.java

```diff
@@ -70,30 +70,49 @@
     this.logger = logger;
   }
 
+  /**
+   *
+   * @param datanodeDetails DatanodeDetails for the DN
+   * @param replicaProto Protobuf representing the replicas
+   * @param publisher EventPublisher instance
+   * @throws IOException
+   * @throws InvalidStateTransitionException
+   */
+  protected void processContainerReplica(final DatanodeDetails datanodeDetails,
+      final ContainerReplicaProto replicaProto, final EventPublisher publisher)
+      throws IOException, InvalidStateTransitionException {
+    ContainerInfo container = getContainerManager().getContainer(
+        ContainerID.valueOf(replicaProto.getContainerID()));
+    processContainerReplica(
+        datanodeDetails, container, replicaProto, publisher);
+  }
+
   /**
    * Process the given ContainerReplica received from specified datanode.
    *
    * @param datanodeDetails DatanodeDetails of the node which reported
    *                        this replica
+   * @param container ContainerInfo represending the container
    * @param replicaProto ContainerReplica
+   * @param publisher EventPublisher instance
    *
    * @throws IOException In case of any Exception while processing the report
    */
   protected void processContainerReplica(final DatanodeDetails datanodeDetails,
-      final ContainerReplicaProto replicaProto, final EventPublisher publisher)
+      final ContainerInfo container, final ContainerReplicaProto replicaProto,
```

Comment:
While you are at it. Nit: `
```suggestion
      final ContainerInfo containerInfo, final ContainerReplicaProto replicaProto,
```